rclpy/rclpy/parameter_service.py

```diff
@@ -71,21 +70,31 @@ def _describe_parameters_callback(self, request, response):
             try:
                 descriptor = node.describe_parameter(name)
             except ParameterNotDeclaredException:
-                descriptor = ParameterDescriptor()
+                response.descriptors = []
+                return response
             response.descriptors.append(descriptor)
         return response
 
     def _get_parameters_callback(self, request, response):
         node = self._get_node()
         for name in request.names:
-            p = node.get_parameter_or(name)
-            response.values.append(p.get_parameter_value())
+            try:
+                param = node.get_parameter(name)
+            except ParameterNotDeclaredException:
+                response.values = []
+                return response
+            response.values.append(param.get_parameter_value())
         return response
 
     def _get_parameter_types_callback(self, request, response):
         node = self._get_node()
         for name in request.names:
-            response.types.append(node.get_parameter_or(name).type_.value)
+            try:
```

Comment:
I think we should provide a `get_parameter_types` method for `node`, and then it will be called here.

Reply:
right, added as you suggested.